### PR TITLE
Update minimum dependencies for 0.20

### PR DESCRIPTION
**Pull request by TomNicholas**
=============== ====== ====
  Package                Old    New
  =============== ====== ====
  cartopy         0.17   0.18
  cftime          1.1    1.2
  dask            2.15   2.30
  distributed     2.15   2.30
  hdf5            1.10   1.12
  lxml            4.5    4.6
  matplotlib-base 3.2    3.3
  numba           0.49   0.51
  numpy           1.17   1.18
  pandas          1.0    1.1
  pint            0.15   0.16
  scipy           1.4    1.5
  seaborn         0.10   0.11
  sparse          0.8    0.11
  toolz           0.10   0.11
  zarr            2.4    2.5
  =============== ====== ====

- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by TomNicholas** [comment]
Oh right I guess I can't just naively update the conda environment like that, because it might not be a valid solve?

**Comment by Illviljan** [comment]
Might be `setuptools`, I had similar errors in #5892.

**Comment by keewis** [comment]
> Might be `setuptools`

if that's the case this should be fixed by merging #5845 first

**Comment by TomNicholas** [comment]
```
Run mamba env update -f $CONDA_ENV_FILE
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache


Looking for: ['python=3.7', 'boto3=1.13', 'bottleneck=1.3', 'cartopy=0.18', 'cdms2=3.1', 'cfgrib=0.9', 'cftime=1.2', 'coveralls', 'dask=2.30', 'distributed=2.30', 'h5netcdf=0.8', 'h5py=2.10', 'hdf5=1.12', 'hypothesis', 'iris=2.4', 'lxml=4.6', 'matplotlib-base=3.3', 'nc-time-axis=1.2', 'netcdf4=1.5.3', 'numba=0.51', 'numpy=1.18', 'pandas=1.1', 'pint=0.16', 'pip', 'pseudonetcdf=3.1', 'pydap=3.2', 'pynio=1.5', 'pytest', 'pytest-cov', 'pytest-env', 'pytest-xdist', 'rasterio=1.1', 'scipy=1.5', 'seaborn=0.11', 'setuptools', 'sparse=0.11', 'toolz=0.11', 'typing_extensions=3.7', 'zarr=2.5']


Encountered problems while solving:
  - package scipy-1.5.0-py36h2d22cac_0 requires python_abi 3.6.* *_cp36m, but none of the providers can be installed
  - package cartopy-0.18.0-py36h91e5944_2 requires python_abi 3.6.* *_cp36m, but none of the providers can be installed

Error: Process completed with exit code 1.
```

What is `python_abi 3.6.* *_cp36m`?

File: ci/requirements/py37-min-all-deps.yml [hunk 1]
```diff
@@ -42,13 +42,13 @@ dependencies:
   - pytest-env
   - pytest-xdist
   - rasterio=1.1
-  - scipy=1.4
-  - seaborn=0.10
+  - scipy=1.5
+  - seaborn=0.11
   # don't need to pin setuptools, now that we don't depend on it
   - setuptools
```
Comment (Illviljan) [inline ci/requirements/py37-min-all-deps.yml:48]:
```suggestion
```

**Comment by Illviljan** [comment]
@TomNicholas Try removing setuptools in `py37-min-all-deps.yml`.

**Comment by dcherian** [comment]
i got the env working but not sure if tests pass

**Comment by Illviljan** [comment]
Maybe if you'd increase scipy to 1.5.2 it would've worked? 

```
(xarray-tests) C:\Users\J.W>conda search conda-forge::scipy=1.5
Loading channels: done
# Name                       Version           Build  Channel
scipy                          1.5.2  py36h7ff6e69_2  conda-forge
scipy                          1.5.2  py37h6db1a17_2  conda-forge
scipy                          1.5.2  py38h5f893b4_2  conda-forge
scipy                          1.5.2  py39hc0c34ad_2  conda-forge
scipy                          1.5.3  py36h27d303f_1  conda-forge
scipy                          1.5.3  py36h7ff6e69_0  conda-forge
scipy                          1.5.3  py37h6db1a17_0  conda-forge
scipy                          1.5.3  py38h5f893b4_0  conda-forge
scipy                          1.5.3  py39hc0c34ad_0  conda-forge

```

File: ci/requirements/py37-min-all-deps.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: xarray-tests
+name: xarray-min-deps
```
Comment (dcherian) [inline ci/requirements/py37-min-all-deps.yml:1]:
```suggestion
name: xarray-tests
```

**Comment by keewis** [comment]
just for reference, apparently `cartopy=0.18` conflicts with `pynio` (which we were planning to move to a separate repository), and `h5py=2.10` seems to be incompatible with `hdf5=1.12` (there is a stalled PR on the feedstock)

**Comment by TomNicholas** [comment]
I'm like staring at this one yellow dot crossing my fingers for you @dcherian lol

**Comment by keewis** [comment]
the download speed for that specific CI job is *really* slow

**Comment by TomNicholas** [comment]
It passed! Well done @dcherian ! Can we merge this then? And then release??

**Comment by dcherian** [comment]
Sounds good to me!

**Pull request merged**